### ISSUE TITLE
New feature: Support API endpoints protected by kerberos

### ISSUE THREAD

**Issue by brijs** [a85/POSTMan-Chrome-Extension#2240]
Postman doesn't seem to support authentication using **kerberos**, or more generally **HTTP Negotiate (SPNEGO)** mechanism.  

The equivalent way of authenticating using kerberos via `curl` is something like below:

``` shell
$ kinit
$ ... # enter password for user
$ curl --negotiate -u : <api_endpoint>
#200 OK
```

**Comment by jaxspades** [comment]
I would love to see this as a part of some of the other authentication types as well. I have endpoints that are using NTLM + Kerberos. I can't seem to get everything working with them using simply NTLM on the native windows client. They used to work with the Chrome App as Chrome was supplying my credentials behind the scenes.

**Comment by Rob-DeRosia** [comment]
Any update on when we can use Windows credentials for authentication purposes? I'd rather not store my username and password with my collection and instead have the application behave like it did with the Chrome app, passing it behind the scenes.

**Comment by ChristopherPope** [comment]
Hi, I've been waiting for Postman to support Kerberos. @a85 , since you self-assigned this ticket can I assume it will be completed soon? 

Thanks.

**Comment by josezeta** [comment]
Is there any deadline or something for this feature?

**Comment by drswin01** [comment]
Is there any update on this feature?

**Comment by Ashthos** [comment]
+1. Some information/updates on this issue would be appreciated.

**Comment by mortenn** [comment]
This is a blocker for our company.

**Comment by cheahkhing** [comment]
Hi, it seems like this has been opened for quite some time without any more updates or progress. Can anyone comment on this?

**Comment by Warkdev** [comment]
Hello folks,

Are you planning to integrate kerberos authentication? That would be handy in several cases. 

Regards.

**Comment by BEagle1984** [comment]
Hi guys. Any news? We badly need it over here.

**Comment by armarti** [comment]
Make a PR. It's open source. Seems like an easy issue to tackle. It's free. Anyone using this is a software developer. If you really really really need postman to handle two WWW-Authenticate headers, make postman handle two WWW-Authenticate headers. This is how you pay for free stuff.

**Comment by BEagle1984** [comment]
> Make a PR. It's open source. Seems like an easy issue to tackle. It's free. Anyone using this is a software developer. If you really really really need postman to handle two WWW-Authenticate headers, make postman handle two WWW-Authenticate headers. This is how you pay for free stuff.

Makes sense, thanks for helping. Except for the fact that we are actually paying a subscription fee for our shared workspace. ;)

**Comment by raoulduke** [comment]
A definite blocker for our work. Just when the team was warming up to Postman we encounter this after I implemented our SSO. Doesn't seem to be in the works or even on the radar, no responses even for the paying customers. Moving back to Restlet/Talend.

**Comment by micheljung** [comment]
@armarti so where's the source code?

**Comment by ghost** [comment]
Darn... RIP Postman 😢

**Comment by prmichaelsen** [comment]
4 years and no dice :(

**Comment by cdavid15** [comment]
Slightly disappointing this still hasn't materialised especially when you launch the chrome app you get presented with the following:

![image](https://user-images.githubusercontent.com/3115533/77992893-e8ca3300-731e-11ea-8ade-298df5413be6.png)


Is there any plans for introducing support for kerberos as this single feature is preventing me from migrating away from the chrome app to the Native app? 

It shouldn't bother me but seeing this popup on launch everyday frustrates me as it is not a true reflection / misleading as the native app has less features than the chrome app.

**Comment by drswin01** [comment]
@micheljung 
https://github.com/postmanlabs/postman-runtime/tree/develop/lib/authorizer
I believe this is where authorization types are implemented for postman but I'm not 100% sure. Hopefully someone who knows a lot more about Node.js than I do can go from here. 
Hope this helps!

**Comment by folprates** [comment]
Hi Guys,

Any update on this?

Thanks,
Fernando

**Comment by DanielBoulerice** [comment]
I am terribly sorry to learn the state of fact concerning Kerberos support in Postman.  I had plans with that in our company.  So I guess it worked in the plug-in because it was hosted in Chrome, is that it?  Is there a way to integrate plug-ins in Postman and put authenticator functionality on a bus?

**Comment by Lonli-Lokli** [comment]
I find it easier to deploy service, returning kerberos token in corporate network based on request, instead of relying on postman roadmap

**Comment by zyystudio** [comment]
how about integrating with spnego kerberos?

**Comment by anddyson** [comment]
Really frustrating that the native app _still_ doesn't support this, when the Chrome one is dead. As others have said, it's a big blocker to using postman for internal testing. This is from a company which might consider buying a subscription, if the tool could actually do something fairly fundamental like this.

**Comment by prusswan** [comment]
> I find it easier to deploy service, returning kerberos token in corporate network based on request, instead of relying on postman roadmap

which led to something like https://github.com/genotrance/px

next step is to look for a standalone/self-contained replacement for Postman

maybe this? https://github.com/hoppscotch/hoppscotch

**Comment by vanosuzume** [comment]
> > I find it easier to deploy service, returning kerberos token in corporate network based on request, instead of relying on postman roadmap我发现部署服务更容易，根据请求在企业网络中返回 kerberos 令牌，而不是依赖邮递员路线图
> 
> which led to something like https://github.com/genotrance/px这导致了类似[https://github.com/genotrance/px 的](https://github.com/genotrance/px)内容
> 
> next step is to look for a standalone/self-contained replacement for Postman下一步是寻找 Postman 的独立/独立替代品
> 
> maybe this? https://github.com/hoppscotch/hoppscotch也许这个？ https://github.com/hoppscotch/hoppscotch

I found a [replacement for postman](https://dev.to/apilover/the-best-free-api-testing-tool-supporting-kerberos-4dec)

maybe there is a better alternative?

**Comment by vanosuzume** [comment]
> > I find it easier to deploy service, returning kerberos token in corporate network based on request, instead of relying on postman roadmap
> 
> which led to something like https://github.com/genotrance/px
> 
> next step is to look for a standalone/self-contained replacement for Postman
> 
> maybe this? https://github.com/hoppscotch/hoppscotch

I found a [replacement for postman](https://dev.to/apilover/the-best-free-api-testing-tool-supporting-kerberos-4dec)

maybe there is a better alternative?

**Comment by raoulduke** [comment]
> Make a PR. It's open source. Seems like an easy issue to tackle. It's free. Anyone using this is a software developer. If you really really really need postman to handle two WWW-Authenticate headers, make postman handle two WWW-Authenticate headers. This is how you pay for free stuff.

FYI, there are many people that pay for Postman (https://www.postman.com/pricing/).

Any updates? Looks like some people have been waiting 9 years...

**Comment by vanosuzume** [comment]
您好，您的邮件我已收到，我会尽快给您回复。

**Comment by prusswan** [comment]
May be possible to use an external library for this:

https://github.com/postmanlabs/postman-app-support/issues/13542
